andela/ah-codeblooded-frontend
# #160577569 Enable users to reset their passwords

**What does this PR do?**

Adds password reset logic. This is triggered by clicking forgot password link on the login or sign up pages. This allows the user to enter an email where the password reset link will be sent. After opening the link the user is taken to the frontend where they key in their new password.

**Description of Task to be completed?**

A user should be presented with a button and or link to reset their passwords on the SignUp/Register component.

**How should this be manually tested?**

This PR can be tested manually by following the following steps on the staging or review app:
1. Create an account.
2. Click Forgot password link on the signup or login form. Route: `/forgot-password`.
3. Enter email address and click send me link button.
4. Go to email and click the password reset link.
5. Enter the email, new password and confirm the password fields.
6. Click reset password button to reset the password. Route: `/reset-password`

**Any background context you want to provide?**

This PR requires the signup and login components to have the Forgot password links.

**What are the relevant pivotal tracker stories?**

[#160577569](https://www.pivotaltracker.com/story/show/160577569)

**Screenshots**
<img width="654" alt="screen shot 2018-11-14 at 20 22 39" src="https://user-images.githubusercontent.com/8180548/48501250-e2ccd280-e84d-11e8-8499-f6eeacf64eb6.png">
<img width="628" alt="screen shot 2018-11-14 at 20 23 24" src="https://user-images.githubusercontent.com/8180548/48501253-e52f2c80-e84d-11e8-9f3b-ee6e9de56019.png">
<img width="629" alt="screen shot 2018-11-14 at 20 27 24" src="https://user-images.githubusercontent.com/8180548/48501258-e6f8f000-e84d-11e8-9491-cd3558c22f1d.png">
<img width="748" alt="screen shot 2018-11-14 at 20 38 12" src="https://user-images.githubusercontent.com/8180548/48501260-e8c2b380-e84d-11e8-8b88-0e2748fc1e4d.png">
<img width="657" alt="screen shot 2018-11-14 at 20 44 17" src="https://user-images.githubusercontent.com/8180548/48501379-258eaa80-e84e-11e8-9fc4-38f9468658ff.png">
